ops/model.py

```diff
@@ -622,3 +628,15 @@ def storage_add(self, name, count=1):
         if not isinstance(count, int) or isinstance(count, bool):
             raise TypeError(f'storage count must be integer, got: {count} ({type(count)})')
         self._run('storage-add', f'{name}={count}')
+
+    def function_get(self):
+        return self._run(f'{self._function_cmd_prefix}-get', return_output=True, use_json=True)
+
+    def function_set(self, results):
+        self._run(f'{self._function_cmd_prefix}-set', *[f"{k}='{v}'" for k, v in results.items()])
```

Comment:
What happens if there are quotes in those values? Ideally function-set can take parameters in the 'foo=bar' style, unquoted, and ideally _run is indeed passing argv arguments into it instead of using the shell. This would indicate the quotes here are unnecessary which is the easiest fix. If either of those assumptions are wrong, this needs a more complex fix.

Reply:
You're right, this definitely needs changing as we're not passing a command to a shell.

I looked at the function-set code again and it uses `strings.SplitN(arg, "=", 2)` so the key is separated from the value based on the first equal sign.

```
juju show-action-output 2              
# ...
results:
  ran: many=equal=signs=in=the=value
status: completed
# ...
```

Thanks for noticing this.